src/app/utils/firestore.ts

```diff
@@ -0,0 +1,62 @@
+import { Course, Question } from "../types/db";
+import { DocumentData, QueryDocumentSnapshot } from "firebase/firestore";
+
+export const courseConverter = {
+  toFirestore(course: Course): DocumentData {
```

Comment:
I think having the type to be `Course` is a bit restricting for end-users, i.e they must always pass in the full-object. So `PartialWithFieldValue<Course>` might make more sense here. Similar comment for `Question`.